### PR TITLE
Feat/more kernels v2 electric boogaloo

### PR DESCRIPTION
**Pull request by db091756**
### PR Type

- Feature
- Tests

### Description

Add Poisson kernel *see Table 3.1 (https://arxiv.org/pdf/1605.09522)). Add Matérn kernel (see https://en.wikipedia.org/wiki/Mat%C3%A9rn_covariance_function), with no explicit gradients.

### How Has This Been Tested?

Added unit tests.

### Does this PR introduce a breaking change?

(Write your answer here.)

### Screenshots

(Write your answer here.)

### Checklist before requesting a review
- [x] I have made sure that my PR is not a duplicate.
- [x] My code follows the style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have performed a self-review of my code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: coreax/kernels/scalar_valued.py [hunk 1]
```diff
@@ -181,6 +182,142 @@ def divergence_x_grad_y_elementwise(self, x: ArrayLike, y: ArrayLike) -> Array:
...
+        body = (jnp.sqrt(2 * self.degree + 1) * norm) / self.length_scale
+        factor = (
+            self.output_scale**2
+            * jnp.exp(-body)
+            * factorial(self.degree)
+            / factorial(2 * self.degree)
+        )
+
+        summation = 1.0
+        if self.degree > 0:
```
Comment (pc532627) [inline coreax/kernels/scalar_valued.py:315]: No test hits the case where self.degree is not greater than 0

File: tests/unit/test_kernels.py [hunk 2]
```diff
@@ -863,6 +865,276 @@ def test_invalid_parameters(self, parameters: tuple, error_msg: str):
             LinearKernel(*parameters)
 
 
+# Cannot inherit from the gradient and mean tests as the domain of the Poisson kernel
+# is not compatible with the data used.
+class TestPoissonKernel(
+    BaseKernelTest[PoissonKernel],
+):
+    """Test ``coreax.kernels.DiracKernel``."""
```
Comment (pc532627) [inline tests/unit/test_kernels.py:873]: DiracKernel -> PoissonKernel

File: tests/unit/test_kernels.py [hunk 3]
```diff
@@ -1177,6 +1449,96 @@ def test_invalid_parameters(self, parameters: tuple, error_msg: str):
...
+            length_scale=parameters[1].item(),
+            degree=int(jnp.ceil(jnp.abs(parameters[2]))) + 1,
+        )
+
+    @pytest.fixture(params=["floats", "vectors", "arrays"])
+    def problem(  # noqa: C901
+        self, request: pytest.FixtureRequest, kernel: MaternKernel
+    ) -> _Problem:
+        r"""
+        Test problems for the SquaredExponential kernel.
```
Comment (pc532627) [inline tests/unit/test_kernels.py:1470]: This and the kernel defined below refer to squared exponential but should refer to Matern

**Review by pc532627 — CHANGES_REQUESTED** [review]
A few minor change requests.

File: coreax/kernels/scalar_valued.py (hunk 1 above)
Comment (db091756) [reply coreax/kernels/scalar_valued.py:315]: This should occur in the mean and gradient tests as the sampled degree parameter is not zero

Comment (db091756) [reply coreax/kernels/scalar_valued.py:315]: I can add analytical tests as well?

Comment (pc532627) [reply coreax/kernels/scalar_valued.py:316]: From a coverage report I don't see it being hit - I think an analytical test would probably be best if possible.

**Review by pc532627 — APPROVED** [review]
